imap/jmap_api.c

```diff
@@ -2254,7 +2254,10 @@ HIDDEN json_t *jmap_copy_reply(struct jmap_copy *copy)
 
 /* Foo/query */
 
-HIDDEN jmap_filter *jmap_buildfilter(json_t *arg, jmap_buildfilter_cb *parse)
+HIDDEN jmap_filter *jmap_buildfilter(jmap_req_t *req,
```

Comment:
Why not plumb `req` through in `rock` in those callbacks that actually need it?

Reply:
Yeah, I can do that